src/workerd/io/worker.c++

```diff
@@ -2420,75 +2536,183 @@ private:
   kj::MutexGuarded<kj::Own<State>> state;
   // Mutex ordering: You must lock this *before* locking the isolate.
 
-  class XThreadNotifier final: public kj::AtomicRefcounted {
-    // Class encapsulating the ability to notify the inspector thread from other threads when
-    // messages are pushed to the outgoing queue.
-    //
-    // TODO(cleanup): This could be a lot simpler if only it were possible to cancel
-    //   an executor.executeAsync() promise from an arbitrary thread. Then, if the inspector
-    //   session was destroyed in its thread while a cross-thread notification was in-flight, it
-    //   could cancel that notification directly.
-  public:
-    void clear() {
-      // Must call in main thread before it drops its reference.
-      paf = nullptr;
+  volatile bool networkEnabled = false;
+  // Not under `state` lock due to lock ordering complications.
+
+  void dispatchProtocolMessage(
+      kj::Locked<kj::Own<workerd::Worker::Isolate::InspectorChannelImpl::State>>& lockedState,
+      kj::String message) {
+    v8_inspector::V8InspectorSession& session = *lockedState->get()->session;
+    // const_cast OK because we're going to lock it
+    Isolate& isolate = const_cast<Isolate&>(*lockedState->get()->isolate);
+    jsg::V8StackScope stackScope;
+    Isolate::Impl::Lock recordedLock(isolate, InspectorLock(nullptr), stackScope);
+    dispatchProtocolMessage(kj::mv(message), session, isolate, stackScope, recordedLock);
+  }
+
+  void dispatchProtocolMessage(kj::String message,
```

Comment:
Looks like the code in this method originates from `incomingLoop()`. Any chance you can move this method up to be in the same place in the class where `incomingLoop()` used to be, so that it's easier to see the diff? We could move it back down here (to be with other `private` stuff) in a separate commit (probably wait to do that until the code is ready to merge, or even do it in a separate PR).

Reply:
Done.